src/operator/matrix_op-inl.h

```diff
@@ -643,6 +654,10 @@ void Slice(const TBlob &src,
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
   int ndim = static_cast<int>(ret->shape_.ndim());
 
+  int axis_size = static_cast<int>(src.shape_[param.axis]);
+  param.begin = (param.begin < 0) ? (param.begin + axis_size + 1) : param.begin;
```

Comment:
Yes, `a[-1:3]` works in numpy. But `a[0:-1]` doesn't give you all values in numpy (it drops the last one). Numpy uses `a[0:]` to take all elements. How can we support such syntax?